### ISSUE TITLE
o-buttons:  low contrast on focus state - ID: DAC_Non_Text_Contrast_05

### ISSUE THREAD

**Issue by JakeChampion** [Financial-Times/origami#504]
# Issue
This is from page 36 of the [audit report](https://drive.google.com/file/d/17wq6b0CIX41RA4muBFDzUDxE0-qzHBIH/view)

Issue can be found on: <https://www.ft.com/__origami/service/build/v3/demo?component=o-buttons%407.0.0&demo=primary&system_code=origami-repo-data&brand=master>

When the ‘Primary’ button receives keyboard focus, the focus indicator fails to meet WCAG AA guidelines for non-text contrast. This issue may affect low-vision users, who may find difficulty in distinguishing the grey focus indicator against the turquoise background colour of the toggle button.
To meet AA guidelines for non-text contrast the minimum required contrast ratio is 3.1 (it is currently 2.1:1).

# Proposed Solution:
Ensure the border colour meets or exceeds the minimum required contrast ratio of 3.1 against adjacent colours in order to satisfy WCAG 2.1 Level AA criteria. This will enable low- vision and keyboard only users to be able to identify where the focus is on the page correctly.

**Comment by JakeChampion** [comment]
I think the part of the codebase which causes the low contrast is this section:
https://github.com/Financial-Times/origami/blob/c1cb152c2e82c51428478628b2e0994da585f639/components/o-buttons/src/scss/_functions.scss#L182

**Comment by notlee** [comment]
@JakeChampion that line is the contrast aim for the background colour between states – not the outline, the outline colour is the same for all states, that comes from o-normalise. We might want to have the focus outline colour change with the button theme/state. Else a different design change that could help is adding an offset between the button and the focus outline. [WCAG have some more good suggestions](https://www.w3.org/WAI/WCAG22/Understanding/focus-appearance-minimum.html).

Brain dump: another complexity is the contrast between the focus outline and the background colour behind the button - which we don't control

**Comment by JakeChampion** [comment]
This issue is also present on <https://www.ft.com/__origami/service/build/v3/demo?component=o-toggle@3.0.0&demo=display-toggle&system_code=origami-repo-data&brand=master> and found on page 201 of the audit report

ID: DAC_Non_Text_Contrast_10

**Comment by notlee** [comment]
https://www.figma.com/file/FLsNMzScj76auCGzEXHFDL/focus-ring?node-id=0%3A1

**Comment by notlee** [comment]
https://github.com/Financial-Times/origami/pull/842
